src/concrete/ml/pytest/torch_models.py

```diff
@@ -1494,6 +1494,27 @@ def predict(x, weights, bias):
         return outputs.squeeze()
 
 
+class WhereNet(torch.nn.Module):
+    """Simple network with a where operation for testing."""
+
+    def __init__(self, n_hidden):
+        super().__init__()
+        self.n_hidden = n_hidden
+        self.fc_tot = torch.rand(1, n_hidden) > 0.5
+
+    def forward(self, x):
+        """Forward pass.
+
+        Args:
+            x (torch.Tensor): The input tensor.
+
+        Returns:
+            torch.Tensor: The output tensor after applying the where operation.
+        """
+        y = torch.where(self.fc_tot, x, 0.0)
```

Comment:
the condition here is not done on the encrypted variable. is this the case we should test ?

Reply:
I am not sure I understand. self.fc_tot is the condition, a constant, not encrypted and x is the input encrypted tensor that will be 0 where condition matches?

Reply:
I think we should also test the case when the condition is on an encrypted variable

Reply:
Isn't this the case? x is encrypted and self.fc_tot isn't. Do you mean "is **not** on an encrypted variable?"

Reply:
Ok sorry you are right the conditions are in clear we are just selecting a part of the encrypted tensor.